test/unit/matrix/test_panel.cpp

```diff
@@ -719,3 +720,49 @@ TYPED_TEST(PanelStoreTransposedTest, OffsetTileUnalignedCol) {
     for (const auto& [size, blocksize, _] : test_params)
       testOffsetTileUnaligned<Coord::Col, TypeParam, StoreTransposed::Yes>(size, blocksize, comm_grid);
 }
+
+struct ConfigSub {
+  const TileElementSize block_size;
+  const GlobalElementIndex sub_origin;
+  const GlobalElementSize sub_size;
+};
+
+const std::vector<ConfigSub> sub_configs{
+    {{3, 4}, {4, 2}, {12, 20}},
+    {{3, 5}, {7, 1}, {12, 11}},
+};
+
+TYPED_TEST(PanelTest, MatrixWithOffset) {
```

Comment:
The test is not very extended, since I didn't want to spend to much time on this considering it will be changed soon. Let me know if you'd like to see it a bit more extended than this.